monday/graphqlclient/client.py

```diff
@@ -15,27 +17,43 @@ def inject_token(self, token, headername='Authorization'):
         self.token = token
         self.headername = headername
 
+    def inject_headers(self, headers):
+        self.headers = headers
+
     def _send(self, query, variables):
         payload = {'query': query}
-        headers = {}
+        headers = self.headers
         files = None
-
+        
         if self.token is not None:
             headers[self.headername] = '{}'.format(self.token)
 
         if variables is None:
-            headers['Content-Type'] = 'application/json'
+            if 'Content-Type' not in headers:
+                headers['Content-Type'] = 'application/json'
```

Comment:
Thanks for the great work @willnaoosmith and very happy to see it was picked up by the authors so quickly. Just a comment:

Mutating the headers seems like a bad idea; headers are not copied at any point in time in this PR, so they are shared across all Resources. 

```python
client = MondayClient(token = "abc",headers = {'API-Version' : '2023-10'})
client.items.client.headers # {'API-Version': '2023-10'}
client.boards.fetch_columns_by_board_id("lala") # Note this is on client.boards not client.items
client.items.client.headers 
""" Output {'API-Version': '2023-10',
 'Authorization': 'abc',
 'Content-Type': 'application/json'}
"""
```

Reply:
Good catch.  This was caught by our internal linter, so I'd glossed over it as a syntactic issue, but this is definitely a genuine bug.